util/fibers/uring_proactor.cc

```diff
@@ -310,12 +314,28 @@ SubmitEntry UringProactor::GetSubmitEntry(CbType cb, uint16_t submit_tag) {
   return SubmitEntry{res};
 }
 
-int UringProactor::RegisterBuffers(const struct iovec* iovecs, unsigned nr_vecs) {
-  return io_uring_register_buffers(&ring_, iovecs, nr_vecs);
+int UringProactor::RegisterBuffers(size_t size) {
+  size = (size + 4096 - 1) / 4096 * 4096;
+  buf_pool_.backing = new (std::align_val_t(4096)) uint8_t[size];
+  buf_pool_.segments.Grow(size / 4096);
+
+  iovec vec{buf_pool_.backing, size};
+  return io_uring_register_buffers(&ring_, &vec, 1);
 }
 
-int UringProactor::UnregisterBuffers() {
-  return io_uring_unregister_buffers(&ring_);
+std::optional<UringBuf> UringProactor::RequestBuffer(size_t size) {
+  DCHECK(buf_pool_.backing);
+  size = (size + 4096 - 1) / 4096 * 4096;
+  if (auto offset = buf_pool_.segments.Request(size / 4096)) {
```

Comment:
please add a comment saying that we use segments to track allocations in 4KB units.
Also, consider introducing a constexpr for 4kb size.